popup/url-withlocation.md

```diff
@@ -0,0 +1,73 @@
+# Dynamically create a hyperlink with lat long values in a popup
+
+This example shows how to create a custom hyperlink based on a hard coded base url in combination with lat and long values calculated using Arcade
+
+## Use cases
+
+* Open Google Maps and drop a pin at the same location
+* Open Google Maps and get directions from your Current Location to the specified location
+* Open Google Maps and open a panoramic view at the same location
+
+This code gets latitude and longitude values for points, lines and polygons using Arcade functions. For lines and polygons, centroid values are generated. As Arcade passes back geometries in the spatial reference of the Web Map basemap, it is likely that a conversion is necessary as most basemaps in ArcGIS Online are in Web Mercator Auxiliary Sphere. The conversion from Web Mercator Auxiliary Sphere to WGS84  is included in the code.
+
+## Workflow
+
+Copy and paste the expression found in the expression template below to the Arcade editor in ArcGIS Online, the relevant location in ArcGIS Pro, or the relevant location in a custom app.
+
+To configure the script to your layer, you replace the EndGoal variable with one of Google Directions, Google Panoramic or Google Pin. If you want to pass these geometry values to other API's, simply alter the code to add a new BaseUrl and pass the appropriate parameters.
+
+## Expression Template
+
+```js
+//Options are Google Directions, Google Panoramic and Google Pin
+var EndGoal = "Google Directions"
+
+var BaseUrl;
+if (EndGoal == "Google Directions") {
+	BaseUrl = "https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination="
+} else if (EndGoal == "Google Panoramic") {
+	BaseUrl = "https://www.google.com/maps/@?api=1&map_action=pano&viewpoint="
+} else if (EndGoal == "Google Pin") {
+	BaseUrl = "https://maps.google.com/maps?t=k&q=loc:"
+} else {
+	Console("Invalid End Goal")
+}
+
```

Comment:
Perhaps we could use [Decode()](https://developers.arcgis.com/arcade/function-reference/logical_functions/#decode) here to condense the code a bit?

So this:

```js
var BaseUrl;
if (EndGoal == "Google Directions") {
	BaseUrl = "https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination="
} else if (EndGoal == "Google Panoramic") {
	BaseUrl = "https://www.google.com/maps/@?api=1&map_action=pano&viewpoint="
} else if (EndGoal == "Google Pin") {
	BaseUrl = "https://maps.google.com/maps?t=k&q=loc:"
} else {
	Console("Invalid End Goal")
}
```

Becomes...

```js
var BaseUrl = Decode( EndGoal,
  "Google Directions", "https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination=",
  "Google Panoramic", "https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=",
  "Google Pin", "https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=",
  "Invalid"
);

if (BaseUrl == "Invalid"){
  Console( EndGoal + " is an invalid value for EndGoal.");
}
```

It looks a little more readable that way, plus the end user gets a more descriptive message when an invalid EndGoal is provided. 

WDYT? Test it out and see if it works OK.